### PR TITLE
Global BranchInputManager

### PR DESCRIPTION
**Pull request by mitchute**
### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/BranchInputManager.cc [hunk 1]
```diff
@@ -67,6 +67,8 @@
 
 namespace EnergyPlus {
 
+BranchInputManagerData dataBranchInputManager;
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc]: New data class that lives here for now.

File: src/EnergyPlus/BranchInputManager.cc [hunk 2]
```diff
@@ -83,88 +85,16 @@ namespace BranchInputManager {
...
     using namespace DataLoopNode;
     using namespace DataBranchAirLoopPlant;
     using namespace NodeInputManager;
     using namespace BranchNodeConnections;
 
-    // Data
     // MODULE PARAMETER DEFINITIONS
     std::string const cMIXER("Connector:Mixer");
     std::string const cSPLITTER("Connector:Splitter");
-    static std::string const BlankString;
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc:106]: All of this now moved to the data class.

File: src/EnergyPlus/BranchInputManager.cc [hunk 3]
```diff
@@ -618,76 +445,45 @@ namespace BranchInputManager {
...
+                BComponents({1, NumComps}) = dataBranchInputManager.Branch(Found).Component({1, NumComps});
             }
         }
     }
 
     void GetNumSplitterMixerInConntrList(std::string const &LoopName,          // Loop Name for this Splitter (used in error message)
                                          std::string const &ConnectorListName, // Requested Connector List Name
-                                         int &NumSplitters,                    // Number of splitters in the loop
-                                         int &NumMixers,                       // Number of mixers in the loop
+                                         int &numSplitters,                    // Number of splitters in the loop
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc:488]: Local variable naming conflict.

File: src/EnergyPlus/BranchInputManager.cc [hunk 4]
```diff
@@ -1049,24 +764,24 @@ namespace BranchInputManager {
...
 
-        if (GetSplitterInputFlag) {
+        if (dataBranchInputManager.GetSplitterInputFlag) {
             GetSplitterInput();
-            GetSplitterInputFlag = false;
+            dataBranchInputManager.GetSplitterInputFlag = false;
         }
 
-        if (ConnectorListName == BlankString) {
+        if (ConnectorListName.empty()) {
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc:778]: ```static std::string const BlankString;``` eliminated and empty strings compared using the ```empty()``` method.

File: src/EnergyPlus/BranchInputManager.cc [hunk 5]
```diff
@@ -1075,9 +790,9 @@ namespace BranchInputManager {
         }
 
         // Default for any errors
-        SplitterName = BlankString;
+        SplitterName = std::string();
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc:799]: Module-level ```BlankString``` usage moved from ```Var = BlankString;``` to ```Var = std::string()``` throughout.

Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc:799]: Module-level ```BlankString``` usage moved from ```Var = BlankString;``` to ```Var = std::string();``` throughout.

File: src/EnergyPlus/BranchInputManager.cc [hunk 6]
```diff
@@ -2599,7 +2181,7 @@ namespace BranchInputManager {
         int IOStat;
 
         // Get Inputs
-        CurrentModuleObject = "CondenserLoop";
+        std::string CurrentModuleObject = "CondenserLoop";
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc:2172]: Moved from module level to the local level.

File: src/EnergyPlus/BranchInputManager.hh [hunk 7]
```diff
@@ -53,49 +53,16 @@
 #include <ObjexxFCL/Optional.hh>
 
 // EnergyPlus Headers
-#include "OutputFiles.hh"
+#include <EnergyPlus/Data/BaseData.hh>
 #include <EnergyPlus/DataGlobals.hh>
 #include <EnergyPlus/DataLoopNode.hh>
 #include <EnergyPlus/EnergyPlus.hh>
+#include <EnergyPlus/OutputFiles.hh>
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.hh:60]: Hat tip to @jasondegraw 🤠

File: src/EnergyPlus/BranchInputManager.hh [hunk 8]
```diff
@@ -112,6 +79,9 @@ namespace BranchInputManager {
         ConnectorData() : NumOfConnectors(0), NumOfSplitters(0), NumOfMixers(0)
         {
         }
+
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.hh:85]: Explicit destructors added for completeness.

File: tst/EnergyPlus/unit/BranchInputManager.unit.cc [hunk 9]
```diff
@@ -85,7 +85,7 @@ TEST_F(EnergyPlusFixture, GetBranchInput_One_SingleComponentBranch)
     ASSERT_TRUE(process_idf(idf_objects));
 
     static std::string const RoutineName("GetBranchInput: ");
-    CurrentModuleObject = "Branch";
+    std::string CurrentModuleObject = "Branch";
```
Comment (mitchute) [inline tst/EnergyPlus/unit/BranchInputManager.unit.cc:88]: No need to have this ```extern```ed and used here. Just making it local.

File: tst/EnergyPlus/unit/BranchInputManager.unit.cc [hunk 10]
```diff
@@ -145,7 +143,6 @@ TEST_F(EnergyPlusFixture, GetBranchInput_One_SingleComponentBranch)
         EXPECT_TRUE(UtilityRoutines::SameString(Alphas(5), "VAV Sys 1 Inlet Node"));
         EXPECT_TRUE(UtilityRoutines::SameString(Alphas(6), "Mixed Air Node 1"));
 
-        NumOfBranches = BCount;
```
Comment (mitchute) [inline tst/EnergyPlus/unit/BranchInputManager.unit.cc:148]: Eliminated a few unused variables.

**Review by mitchute — COMMENTED** [review]
Ready for review.

**Comment by Myoldmopar** [comment]
I have no concerns with this.  It has been nice to see it get fixed up over the commits, and I appreciate the walkthrough of the changes.  If something had been broken CI definitely would've complained like it did in earlier commits.  It's all so green now though...great work forging ahead with it.  I know getting this fleshed out to all these files was a big lift but will get better as these PRs keep coming.  I'm merging this, knowing that it may conflict other branches, but I'd need to pull develop into them while testing anyway.

**Pull request merged**